lib/database/migrations/20240424092020-add-qc-flag-effective-period-columns.js

```diff
@@ -0,0 +1,248 @@
+        ;
+    `, { transaction });
+
+    // Shrink effective periods that end after new flag's period
+    await queryInterface.sequelize.query(`
+        UPDATE quality_control_flag_effective_periods as qcfep
+        ${joinClause}
+        SET qcfep.\`from\` = '${timestampToMysql(intersectingPeriodTo)}'
+        WHERE unix_timestamp(qcfep.\`to\`) * 1000 > ${intersectingPeriodTo}
+        ;
+    `, { transaction });
+};
+
+/**
+ * Assign appropriate effectivePeriods to all QC flags
+ * @param {QueryInterface} queryInterface sequelize interface
+ * @return {Promise<void>} promise
+ */
+const _reconstructQcFlagsEffectivePeriods = async (queryInterface, { transaction }) => {
+    const [qcFlagsPerDataPass] = await queryInterface.sequelize.query(`
+        SELECT 
+            \`from\`,
+            \`to\`,
+            qcf.created_at,
+            run_number,
+            dpl_detector_id,
+            dpqcf.data_pass_id
+        FROM quality_control_flags as qcf
+        INNER JOIN data_pass_quality_control_flag as dpqcf
+            ON qcf.id = dpqcf.quality_control_flag_id
+        ORDER by qcf.created_at ASC
+        ;
+    `, { transaction });
+    for (const flag of qcFlagsPerDataPass) {
+        const {
+            from,
+            to,
+            created_at: createdAt,
+            run_number: runNumber,
+            dpl_detector_id: dplDetectorId,
+            data_pass_id: dataPassId,
+        } = flag;
+        await _updateEffectivePeriodForScopeAndPeriod(
+            queryInterface,
+            { from: from.getTime(), to: to.getTime() },
+            { runNumber, dplDetectorId, dataPassId, createdAt: createdAt.getTime() },
+            { transaction },
+        );
+    }
+
+    const [qcFlagsPerSimulationPass] = await queryInterface.sequelize.query(`
+        SELECT
+            \`from\`,
+            \`to\`,
+            qcf.created_at,
+            run_number,
+            dpl_detector_id,
+            spqcf.simulation_pass_id
+        FROM quality_control_flags as qcf
+        INNER JOIN simulation_pass_quality_control_flag as spqcf
+            ON qcf.id = spqcf.quality_control_flag_id
+        ORDER by qcf.created_at ASC
+        ;
+    `, { transaction });
+
+    for (const flag of qcFlagsPerSimulationPass) {
+        const {
+            from,
+            to,
+            created_at: createdAt,
+            run_number: runNumber,
+            dpl_detector_id: dplDetectorId,
+            simulation_pass_id: simulationPassId,
+        } = flag;
+        await _updateEffectivePeriodForScopeAndPeriod(
+            queryInterface,
+            { from: from.getTime(), to: to.getTime() },
+            { runNumber, dplDetectorId, simulationPassId, createdAt: createdAt.getTime() },
+            { transaction },
+        );
+    }
+};
+
+/**
+ * Create effective period for every existing QC flag over its whole period
+ * @param {QueryInterface} queryInterface sequelize interface
+ * @return {Promise<void>} promise
```

Comment:
What does the promise do? Simply stating promise is worthless...
`Resolves once...`